### PR TITLE
feat: multi-chain walletconnect v1

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Updates the wallet connect v1 implementation to support all EVM chains that Wallet Connect V1 and our Native wallet supports (Ethereum, Avalanche, BNB Chain, Optimism, and Polygon).

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A, though it might prove useful as a fallback for foxatars on mercle, if needed.

## Testing

We can't currently test this on https://app.mercle.xyz/shapeshift due to a wagmi
issue in their implementation.

It also seems the current state of their app is a partial implementation of Wallet
Connect V2, so it can't be tested right now anyway.

We can, however, use OpenSea to confirm this works:

1. Go to https://opensea.io/
2. Connect via WalletConnect (it'll use a WalletConnect V1 URI), which means we ca
n only have one chain connected at a time
3. It'll be connected to Ethereum by default. Clicke the wallet balance in the top
 right, click "Switch Chain", then select a chain that we support on the Native Wa
llet (Ethereum, Avalanche, BNB chain, Optimism, or Polygon)
4. Ensure that the connected chain in the ShapeShift app updated
5. Ensure that we can sign transactions on the new chain. For a cheap NFT to buy, try something like https://opensea.io/assets/matic/0xba6666b118f8303f990f3519df07e160227cce87/8 on Polygon.

Screenshot 2023-05-29 at 12 04 24 pm

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

Screenshot 2023-05-29 at 1 28 16 pm

Screenshot 2023-05-29 at 1 26 31 pm

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4590** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4590" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4590?utm_source=stack-comment).

File: src/context/WalletProvider/WalletProvider.tsx [hunk 1]
```diff
@@ -651,6 +651,11 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
               /** List of RPC URLs indexed by chain ID */
               rpc: {
                 1: getConfig().REACT_APP_ETHEREUM_NODE_URL,
+                10: getConfig().REACT_APP_OPTIMISM_NODE_URL,
+                56: getConfig().REACT_APP_BNBSMARTCHAIN_NODE_URL,
+                100: getConfig().REACT_APP_GNOSIS_NODE_URL,
+                137: getConfig().REACT_APP_POLYGON_NODE_URL,
+                43114: getConfig().REACT_APP_AVALANCHE_NODE_URL,
```
Comment (woodenfurniture) [inline src/context/WalletProvider/WalletProvider.tsx:658]:
```suggestion
                [CHAIN_REFERENCE.EthereumMainnet]: getConfig().REACT_APP_ETHEREUM_NODE_URL,
                [CHAIN_REFERENCE.Optimism]: getConfig().REACT_APP_OPTIMISM_NODE_URL,
                [CHAIN_REFERENCE.BnbSmartChainMainnet]: getConfig().REACT_APP_BNBSMARTCHAIN_NODE_URL,
                [CHAIN_REFERENCE.GnosisMainnet]: getConfig().REACT_APP_GNOSIS_NODE_URL,
                [CHAIN_REFERENCE.PolygonMainnet]: getConfig().REACT_APP_POLYGON_NODE_URL,
                [CHAIN_REFERENCE.AvalancheCChain]: getConfig().REACT_APP_AVALANCHE_NODE_URL,
```

File: src/context/WalletProvider/WalletProvider.tsx [hunk 2]
```diff
@@ -651,6 +651,11 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
               /** List of RPC URLs indexed by chain ID */
```
Comment (woodenfurniture) [inline src/context/WalletProvider/WalletProvider.tsx:651]:
```suggestion
              /** List of RPC URLs indexed by chain reference */
```

File: src/plugins/walletConnectToDapps/v1/WalletConnectBridgeProvider.tsx [hunk 3]
```diff
@@ -219,6 +249,39 @@ export const WalletConnectBridgeProvider: FC<PropsWithChildren> = ({ children })
     [handleDisconnect],
   )
 
+  const handleSwitchChain = useCallback(
+    (err: Error | null, payload: any) => {
+      if (err) return moduleLogger.error(err, 'handleSwitchChain')
+      if (!wcAccountId) return moduleLogger.error('No account id found for wallet connect')
+      moduleLogger.info('handleSwitchChain', payload)
+      const chainIdHex = payload.params[0].chainId
```
Comment (woodenfurniture) [inline src/plugins/walletConnectToDapps/v1/WalletConnectBridgeProvider.tsx:257]: we should make this slightly clearer so downstream code is less confusing for us later (e.g on this line `chainId: chainIdHex, // chain reference as hex`)

```
      const walletConnectChainIdHex = payload.params[0].chainId
```

or 

```
      const chainReferenceHex = payload.params[0].chainId
```

**Review by woodenfurniture — APPROVED** [review]
small comments, pre-approval for when those are resolved

File: src/context/WalletProvider/WalletProvider.tsx (hunk 1 above)
Comment (0xApotheosis) [reply src/context/WalletProvider/WalletProvider.tsx]: 100%. Addressed in https://github.com/shapeshift/web/pull/4590/commits/b6c2b83bf461c9ffc074882fe7a79a8f8d98e320.

File: src/context/WalletProvider/WalletProvider.tsx (hunk 2 above)
Comment (0xApotheosis) [reply src/context/WalletProvider/WalletProvider.tsx]: gm: https://github.com/shapeshift/web/pull/4590/commits/678b074201432aae9fb2db6a10d64424a8e0aaab

File: src/plugins/walletConnectToDapps/v1/WalletConnectBridgeProvider.tsx (hunk 3 above)
Comment (0xApotheosis) [reply src/plugins/walletConnectToDapps/v1/WalletConnectBridgeProvider.tsx]: https://github.com/shapeshift/web/pull/4590/commits/678b074201432aae9fb2db6a10d64424a8e0aaab

**Pull request merged**